### PR TITLE
App mesh

### PR DESCRIPTION
**Pull request by geremyCohen**
https://github.com/aws-samples/eks-workshop/issues/240

First version of App Mesh (Preview version) chapter.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: content/servicemesh_with_appmesh/components.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+draft: false
+---
+
+![App Mesh Architecture](/images/app_mesh/appmesh_overview.png)
+
+App Mesh is made up of the following components:
+
+* Service mesh: A service mesh is a logical boundary for network traffic between the services that reside within it. For more information, see [Service Meshes](https://docs.aws.amazon.com/app-mesh/latest/userguide/meshes.html).
+
+* Virtual nodes: A virtual node acts as a logical pointer to a particular task group, such as an ECS service or a Kubernetes deployment. When you create a virtual node, you must specify the DNS service discovery name for your task group. For more information, see [Virtual Nodes](https://docs.aws.amazon.com/app-mesh/latest/userguide/virtual_nodes.html).
```
Comment (christopherhein) [inline content/servicemesh_with_appmesh/components.md]: should this be `or a Kubernetes Service` not a `or a Kubernetes deployment`?

File: content/servicemesh_with_appmesh/create_virtual_nodes.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+draft: false
+---
+
+Next, we'll create five Virtual Nodes, one for each of the microservices in our application.
+
+### More about Virtual Nodes
+
+A virtual node acts as a logical pointer to a k8s service.
+
+Service Discovery / DNS name of the virtual node is defined in the serviceDiscovery.dns attribute.
```
Comment (christopherhein) [inline content/servicemesh_with_appmesh/create_virtual_nodes.md]: should `serviceDiscovery.dns` have backticks?

File: content/servicemesh_with_appmesh/create_virtual_nodes.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+Inbound traffic parameters for the virtual node are specified in the listener attribute.
+
+Outbound traffic the virtual node forwards to should be specified in the backend attribute.
+
+### colorgateway-vn
+
+We'll first define the virtual node colorgateway-vn.  colorgateway-vn will be the entrypoint to our app.
```
Comment (christopherhein) [inline content/servicemesh_with_appmesh/create_virtual_nodes.md]: wrap `colorgateway-vn` in back ticks

**Comment by christopherhein** [comment]
@geremyCohen and @brentley do we think we can get this updated or rather rebased and merged tonight?

**Comment by geremyCohen** [comment]
package-lock.json restored from master @brentley

**Comment by christopherhein** [comment]
Nice changes @geremyCohen, @brentley will you give this a scan and merge please?

**Pull request merged**